pkg/stub/psmdb.go

```diff
@@ -110,29 +110,25 @@ func (h *Handler) addSpecDefaults(m *v1alpha1.PerconaServerMongoDB) {
 		if spec.Backup.Coordinator == nil {
 			spec.Backup.Coordinator = &v1alpha1.BackupCoordinatorSpec{}
 		}
-		if spec.Backup.S3 == nil {
-			spec.Backup.S3 = &v1alpha1.BackupS3Spec{}
-		}
-		if spec.Backup.S3.Secret == "" {
-			spec.Backup.S3.Secret = backup.DefaultS3SecretName
+		if spec.Backup.Coordinator.EnableClientsLogging == nil {
+			spec.Backup.Coordinator.EnableClientsLogging = backup.DefaultEnableClientsLogging
 		}
 		for _, bkpTask := range spec.Backup.Tasks {
 			if bkpTask.CompressionType == "" {
 				bkpTask.CompressionType = backup.DefaultCompressionType
 			}
-			if bkpTask.DestinationType == "" {
-				bkpTask.DestinationType = backup.DefaultDestinationType
-			}
 		}
 	}
 }
 
 // hasBackupsEnabled returns a boolean reflecting if there are any backups
 // enabled in the PSMDB spec
 func (h *Handler) hasBackupsEnabled(m *v1alpha1.PerconaServerMongoDB) bool {
-	for _, backupTask := range m.Spec.Backup.Tasks {
-		if backupTask.Enabled && backupTask.Schedule != "" {
-			return true
+	if m.Spec.Backup != nil && m.Spec.Backup.Enabled {
+		for _, backupTask := range m.Spec.Backup.Tasks {
+			if backupTask.Enabled && backupTask.Schedule != "" {
```

Comment:
In general, since the object has `Enabled` field,  there is no need to use a pointer to the struct instead of the struct by itself. Here pointer gives no pros but only cons because of necessity for the nil-checks.
But I see no need to change it now. Just a side notes.

Reply:
Good point! I like this idea